app/src/org/odk/collect/android/views/HierarchyElementView.java

```diff
@@ -19,12 +19,14 @@ public HierarchyElementView(Context context, HierarchyElement it) {
 
         RelativeLayout layout = (RelativeLayout)inflate(context, R.layout.hierarchy_element_view, null);
 
-        setBackgroundColor(it.getColor());
+        setBackgroundColor(it.getBgColor());
 
         mPrimaryTextView = ((TextView)layout.findViewById(R.id.hev_primary_text));
         mPrimaryTextView.setText(it.getPrimaryText());
+        mPrimaryTextView.setTextColor(it.getTextColor());
         mSecondaryTextView = ((TextView)layout.findViewById(R.id.hev_secondary_text));
         mSecondaryTextView.setText(it.getSecondaryText());
```

Comment:
Shouldn't this (and above) use setPrimaryText() and setTextcolor() respectively?